torchtune/training/_distributed.py

```diff
@@ -718,3 +743,108 @@ def prepare_mha_for_tp(
     if is_fusion_model:
         model.decoder = decoder
     return model
+
+
+def _get_sdpa_context() -> (
```

Comment:
Does this mean CP doesn't work with FlexAttention?

Reply:
Yes, at least until https://github.com/pytorch/pytorch/pull/151497 lands

Reply:
But I also think this is somewhat orthogonal. Like flex does not have its own backend (see [here](https://github.com/pytorch/pytorch/blob/172a7c942edd4394f81fd0e889f10822af916b9e/aten/src/ATen/SDPBackend.h#L7)). My assumption is that it should be using the flash attention backend (but need to confirm)

Reply:
curious if torchtune already has non-CP flex_attention? cc @ebsmothers

Reply:
@XilunWu yes we do, see e.g. [here](https://github.com/pytorch/torchtune/blob/dedc209c503b9ddcc3dbb62aeb5ae0dd9093b46f/torchtune/modules/attention_utils.py#L195-L227) for how we apply it for sample packing